File: Easy/Hash Tables - Ransom Note/checkMagazine.py
def checkMagazine(magazine, note):
    mag_dict = {}
    note_dict = {}

    for i in magazine:
        if i not in mag_dict:
            mag_dict[i] = 1
        else:
            mag_dict[i] +=1
    
    for i in note:
        if i not in note_dict:
            note_dict[i] = 1
        else:
            note_dict[i] += 1

    for i in note_dict:
        if i not in mag_dict:
            print('No')
            return

        if note_dict[i] > mag_dict[i]:
            print('No')
            return

    print('Yes')

File: Easy/Hash Tables - Ransom Note/checkMagazine.js
function readLine() {
    return inputString[currentLine++];
}

// Complete the checkMagazine function below.
function checkMagazine(magazine, note) {
    let mag_dict = {}
    let note_dict = {}

    for (let i of magazine) {
        if (!mag_dict[i]) {
            mag_dict[i] = 1
        } else {
            mag_dict[i] += 1
        }
    }

    for (let i of note) {
        if (!note_dict[i]) {
            note_dict[i] = 1
        } else {
            note_dict[i] += 1
        }
    }

    for (let i in note_dict){
        if (!mag_dict[i]){
            console.log('No')
            return
        }

        if (note_dict[i] > mag_dict[i]) {
            console.log('No')
            return
        }
    }

    console.log('Yes')
}